ambry-network/src/test/java/com/github/ambry/network/LocalRequestResponseChannelTest.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2021 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.network;
+
+import com.github.ambry.clustermap.MockReplicaId;
+import com.github.ambry.clustermap.ReplicaType;
+import com.github.ambry.network.LocalRequestResponseChannel.LocalChannelRequest;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+public class LocalRequestResponseChannelTest {
```

Comment:
Are there specific tests with leak detection enabled where you have verified the ref count logic?

Reply:
For verifying the ref count logic, I ran test cases in `CloudRouterTest` with leak detection enabled. It seems to exercise putting and getting blobs from `LatchBasedInMemoryCloudDestination` via `LocalNetworkClient`.